### ISSUE TITLE
Expand documentation

### ISSUE THREAD

**Issue by emoszkowski** [JuliaParallel/DistributedArrays.jl#115]
We are finding the documentation for DistributedArrays to be lacking in a few ways: 

1. There is no description of the actual DistributedArray type. It would be helpful to know how DistributedArrays are stored across processes. A lot of people who are using DistributedArrays are experienced at distributed programming, and will find it helpful to know how things are implemented. For instance, it would be helpful to get an idea (without looking at the source code) of how the local parts are tied together in a single type.

2. When are values copied/not copied? For example:
```julia
da = dzeros(100,100)
sa = da[1:10, 35:50]
out = convert(DArray, sa)
```
Does the last line here copy the values of `sa`? More generally, information about performance would be really useful, as well as some performance tips.

3. Expound a bit more on DArray constructor. Perhaps an example somewhere between `dfill` and the Game of Life would help, like this:

```julia
procs = addprocs(5)

# Construct a DArray 
da = DArray((2,5), procs, [1,5]) do I  
	# construct a local part
	localpart = Array{Float64}(2,1)	
	for i in first(I)  # first(I) = 1:2
		localpart[i] =  i
	end
	return localpart
end
```

It would help to describe this example as follows: 

This DArray is distributed so that each of 5 processes holds one column. `I` is a tuple of indices describing the indices of the DArray that reside on a given process. On the first process, `I = (1:2,1:1)`.  In the `init` function, we construct the local column as a regular `Array` and return it to the `DArray` constructor. Note that it is important to return `localpart`: if we don't do this, then Julia returns the last variable that was assigned, which is `localpart[2]` - but the `DArray` constructor is expecting an array.

cc @pearlzli

**Comment by Alexander-Barth** [comment]
Yes, I think that this would be quite helpful. Another example, that might be useful is this function that smooths a distributed vector:

```julia
addprocs(5)
using DistributedArrays
function smoothing(y::DArray)
    return DArray(size(y),procs(y)) do I
        # tmp is the local part
        tmp = zeros(length(I[1])); 
        for i = I[1]
            if i > 1 && i < length(y)
                tmp[i - first(I[1]) + 1] = 0.1 * y[i-1] + 0.8 * y[i] + 0.1 * y[i+1] ;
            else
                tmp[i - first(I[1]) + 1] = y[i]
            end
        end
        return tmp
    end    
end
y = drandn(10)
ys = smoothing(y)
```
